arangod/Scheduler/Scheduler.cpp

```diff
@@ -347,15 +347,18 @@ bool Scheduler::shouldQueueMore() const {
   return false;
 }
 
-bool Scheduler::hasQueueCapacity() const {
-  if (!shouldQueueMore()) {
-    return false;
+bool Scheduler::shouldExecuteDirect() const {
+  uint64_t const counters = _counters.load();
+  uint64_t const nrWorking = numWorking(counters);
+  uint64_t const nrBlocked = numBlocked(counters);
+  
+  if (nrWorking + nrBlocked + _nrQueued < numRunning(counters) / 2 + 1) {
```

Comment:
I think this is a behaviour change, maybe we should review this formula (also in devel !)

Reply:
@graetzer ok, let's do this today